### PR TITLE
DR-2579 Look in the correct place to decide not to error project delete step

### PR DESCRIPTION
**Pull request by nmalfroy**
This wasn't getting caught.

Reproed locally

### CODE REVIEW

**Review by snf2ye — APPROVED** [review]
Thanks!

**Review by okotsopoulos — APPROVED** [review]

**Review by tlangs — APPROVED** [review]

File: src/main/java/bio/terra/service/resourcemanagement/google/GoogleResourceManagerService.java [hunk 1]
```diff
@@ -94,7 +94,11 @@ public void deleteProject(String googleProjectId) {
       // the response will be empty if the request is successful in the delete
       request.execute();
     } catch (IOException | GeneralSecurityException e) {
-      if (e.getMessage().equals("Cannot delete an inactive project.")) {
+      if (e instanceof GoogleJsonResponseException
```
Comment (pshapiro4broad) [inline src/main/java/bio/terra/service/resourcemanagement/google/GoogleResourceManagerService.java:97]: `catch` already does class matching so it seems awkward to use `instanceof` inside of it. Although that would result in a duplicate `throw`.. up to you to decide which is worse

**Comment by pshapiro4broad** [comment]
This seems like something that we'd want an integration test for, in case the error message or error type ever changes.

File: src/main/java/bio/terra/service/resourcemanagement/google/GoogleResourceManagerService.java (hunk 1 above)
Comment (nmalfroy) [reply src/main/java/bio/terra/service/resourcemanagement/google/GoogleResourceManagerService.java:97]: It doesn't...note that `GoogleJsonResponseException` is a subclass

**Comment by nmalfroy** [comment]
Yeah, the problem is that it requires fiddling with the project from the outside while the flight is in process

File: build.gradle [hunk 2]
```diff
@@ -482,11 +482,11 @@ test {
         includeCategories 'bio.terra.common.category.Unit'
     }
     outputs.upToDateWhen { false }
-    retry {
-        if (isCiServer) {
-           maxRetries = 3
-        }
-    }
+//    retry {
```
Comment (okotsopoulos) [inline build.gradle]: Can you remind me why we're testing out no test retries?

**Comment by nmalfroy** [comment]
Oh...shoot,  disabled that locally.  I'll re-add

File: src/main/java/bio/terra/service/resourcemanagement/google/GoogleResourceManagerService.java (hunk 1 above)
Comment (pshapiro4broad) [reply src/main/java/bio/terra/service/resourcemanagement/google/GoogleResourceManagerService.java:97]: I don't understand, wouldn't adding a catch before this one allow you to catch the exception subclass?

File: src/main/java/bio/terra/service/dataset/flight/LockDatasetStep.java [hunk 3]
```diff
@@ -58,7 +59,7 @@ public StepResult doStep(FlightContext context) {
       } else {
         return new StepResult(StepStatus.STEP_RESULT_FAILURE_FATAL, notFoundEx);
       }
-    } catch (RetryQueryException | DatasetLockException e) {
+    } catch (RetryQueryException | DatasetLockException | TransactionSystemException e) {
```
Comment (pshapiro4broad) [inline src/main/java/bio/terra/service/dataset/flight/LockDatasetStep.java:62]: Why are the retryable exceptions in the undo method different from the retryable ones in the do method (and in the unlock step's do method)?

Comment (nmalfroy) [reply src/main/java/bio/terra/service/dataset/flight/LockDatasetStep.java:62]: Ah derp, you're totally right.  I'll update in a followon

**Pull request merged**